### PR TITLE
More accurate error messages for "undefined" built-ins

### LINKED ISSUE #1500: More accurate error when redefining the "undefined" `_NARG` and `@`
Noticed in [this comment](https://github.com/gbdev/rgbds/pull/1498#discussion_r1752601664): `DEF(_NARG)` is false outside of macros, but you can't define or purge it yourself:

```console
% rgbasm - <<EOF
println DEF(_NARG)
def _NARG equ 0
EOF
$0
error: <stdin>(2):
    '_NARG' already defined at <builtin>
error: Assembly aborted (1 error)!
```

We could have a more accurate error message for built-ins here. (The same applies to `@` outside of sections.)

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #1500

### CODE REVIEW

File: src/asm/symbol.cpp [hunk 1]
```diff
@@ -103,6 +103,29 @@ static void updateSymbolFilename(Symbol &sym) {
...
+		fputs(" at ", stderr);
+		dumpFilename(sym);
+	}
+}
+
+static void redefinedError(Symbol const &sym) {
+	assume(sym.isBuiltin);
+	if (!sym_FindScopedValidSymbol(sym.name)) {
+		// `DEF()` would return false, so we should not imply the symbol is already defined
+		error("'%s' is reserved for a built-in symbol\n", sym.name.c_str());
```
Comment (ISSOtm) [inline src/asm/symbol.cpp:123]: I think it's fine to always print this message, since the *name* itself is reserved, whether the symbol is defined or not.

**Review by ISSOtm — DISMISSED** [review]
The refactoring feels good. ^^

File: src/asm/symbol.cpp (hunk 1 above)
Comment (Rangi42) [reply src/asm/symbol.cpp:123]: It would be *fine* to, but this is also fine IMO, and doesn't disturb any test cases (so it's a smaller change overall).

**Review by ISSOtm — APPROVED** [review]
Sure. I would appreciate simpler logic, but your argument has value too. Let's skip the bikeshedding for once :)

**Pull request merged**